src/Nethermind/Nethermind.Core/Collections/IListExtensions.cs

```diff
@@ -124,5 +124,19 @@ private static bool TryGetSearchedItem<T>(this IList<T> list, int index, out T?
                 }
             }
         }
+
+        public static int FindIndex<T>(this IReadOnlyList<T> list, Predicate<T> match, int startIndex = 0, int? endIndex = null)
+        {
+            endIndex ??= list.Count - startIndex;
```

Comment:
Why `- startIndex`?

Reply:
looks like a bug, but I think I refactored code to not use this method so I will just delete it for now.